app/javascript/mastodon/actions/compose.js

```diff
@@ -214,8 +215,8 @@ export function submitCompose(routerHistory) {
         'Idempotency-Key': getState().getIn(['compose', 'idempotencyKey']),
       },
     }).then(function (response) {
-      if (routerHistory && (routerHistory.location.pathname === '/publish' || routerHistory.location.pathname === '/statuses/new') && window.history.state) {
-        routerHistory.goBack();
+      if (browserHistory && (browserHistory.location.pathname === '/publish' || browserHistory.location.pathname === '/statuses/new') && window.history.state) {
```

Comment:
I think that `browserHistory` will always be there. This is to check if we are in a `<Router>` and you can not know it from the action, as it depends on the context's presence when the action is called

Reply:
Ah, indeed, this is a good point. Though I think the only case where a composer would be mounted without a router is in `/share`, so the `pathname` check would already catch this.